File: src/lightnap-ng/src/app/core/backend-api/dtos/external-logins/request/search-external-logins-request-dto.ts
import { PagedRequestDto } from "../../paged-request-dto";

/**
 * Interface representing a request to search for external logins with optional filtering criteria.
 * Extends the PagedRequestDto interface to include pagination properties.
 */
export interface SearchExternalLoginsRequestDto extends PagedRequestDto {
    /**
     * Optional filter for the login provider (e.g., Google, Facebook).
     */
    loginProvider?: string;
    /**
     * Optional filter for the user ID associated with the external login.
     */
    userId?: string;
}

File: src/LightNap.Core/Identity/Dto/Request/SearchExternalLoginsRequestDto.cs
using LightNap.Core.Api;

namespace LightNap.Core.Identity.Dto.Request
{
    /// <summary>
    /// Represents a request to search external logins.
    /// </summary>
    public class SearchExternalLoginsRequestDto : PagedRequestDtoBase
    {
        /// <summary>
        /// The optional login provider to filter by.
        /// </summary>
        public string? LoginProvider { get; set; }

        /// <summary>
        /// The optional user ID to filter by.
        /// </summary>
        public string? UserId { get; set; }
    }
}